File: experimentos/experimento/bm/operaciones/consultaMovimientos.js
import { Estado } from "../estado.js";
import ClaveIncorrectaError from '../errores/ClaveIncorrectaError.js';
import UsuarioInexistenteError from '../errores/UsuarioInexistenteError.js';
import UsuarioNoHabilitadoError from '../errores/UsuarioNoHabilitadoError.js';

const consultaMovimientos = (args) => {
  const [_, dniAdmin, claveAdmin, dniConsulta, desde, hasta] = args;

  const estado = Estado.cargar();

  // Validaciones
  if (dniAdmin !== Estado.ADMINISTRADOR) {
    throw new UsuarioNoHabilitadoError();
  }
  
  if (estado.claves[dniAdmin] !== claveAdmin) {
    throw new ClaveIncorrectaError();
  }

  if (!estado.usuarios[dniConsulta]) {
    throw new UsuarioInexistenteError();
  }

  const movimientos = estado.movimientos[dniConsulta].filter((movimiento) => new Date(movimiento.fechaHora) >= new Date(desde) && new Date(movimiento.fechaHora) <= new Date(hasta));

  return movimientos;
};

export default consultaMovimientos;

File: solucion_modelo/operaciones/movimientos/consulta_movimientos.py
from operaciones.movimientos.parametros_consulta_movimientos import (
    ParametrosConsultaMovimientos,
)
from tipos import RESULTADO
from utiles import es_administrador
from estado import Estado


def consulta_movimientos(solicitud: ParametrosConsultaMovimientos):
    if not es_administrador(solicitud.dni, solicitud.clave):
        return RESULTADO.UsuarioNoHabilitado

    estado = Estado()

    if solicitud.dni_consulta not in estado.usuarios.keys():
        return RESULTADO.UsuarioInexistente

    movimientos = filter(
        lambda m: m.fechahora >= solicitud.desde
        and m.fechahora <= solicitud.hasta
        and m.dni == solicitud.dni_consulta,
        estado.movimientos,
    )

    print(f"Movimientos del usuario DNI {solicitud.dni_consulta}")
    for movimiento in movimientos:
        fecha = movimiento.fechahora.strftime("%Y-%m-%d-%H:%M")
        operacion = movimiento.operacion
        print(f"* {fecha} | {operacion}")

    return RESULTADO.OK
